src/utils/ConditionalWrapper.tsx

```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+
+/**
+ * Utility component for conditional wrapping.
+ */
+export const ConditionalWrapper: React.FC<{
+  condition: unknown
+  wrapper: (children: React.ReactElement) => React.ReactElement
```

Comment:
Could be `children: React.ReactChildren` to be more general

Reply:
The children prop could be optional.

Reply:
I had `React.ReactNode` originally but the typescript complained about undefined. Will try `ReactChildren`.

Reply:
I cant seem to get it to compile with ReactChildren. I guess if it return the unwrapped version you could get an array of react components and not a valid component. Like unwrapped in Fragment.

I've got it to work with ReactNode, but it causes problems downstream dealing with the undefined case. So I think the current version is the best we can do.

Maybe it shouldn't be a FC and would work with something different.

Ah, writing this comment has given me an idea...

Reply:
Plot thickens, This version run the tests fine locally but they fail in the build.